map/src/context/TracksManager.js

```diff
@@ -71,12 +99,240 @@ function prepareName(name, local) {
+function addTrack(ctx, track) {
+    addDistance(track);
+    ctx.localTracks.push(track);
+    ctx.setLocalTracks([...ctx.localTracks]);
+    TracksManager.saveTracks(ctx.localTracks);
+}
+
+function getTrackPoints(track) {
+    let points = [];
+    if (track.tracks) {
+        track.tracks.forEach(track => {
+            track.points.forEach(point => {
+                if (point.geometry) {
+                    point.geometry.forEach(trk => {
+                        points.push(trk);
+                    })
+                } else {
+                    points.push(point);
+                }
+            })
+        })
+    }
+    return points;
+}
+
+function getActivePoints(track) {
+    let points = [];
+    if (track.tracks) {
+        track.tracks.forEach(track => {
+            track.points.forEach(point => {
+                points.push(point);
+            })
+        })
+    }
+    return points;
+}
+
+function addDistance(track) {
+    if (track.tracks) {
+        track.tracks.forEach(t => {
+            if (!track.hasOnlyTrk) {
+                let distanceFromStart = 0;
+                for (let point of t.points) {
+                    if (point.geometry) {
+                        if (point.geometry.length > 0) {
+                            point.geometry.forEach(trk => {
+                                let currIndex = point.geometry.indexOf(trk);
+                                if (trk.distance === 0 && currIndex !== 0) {
+                                    trk.distance = Utils.getDistance(trk.lat, trk.lng, point.geometry[currIndex - 1].lat, point.geometry[currIndex - 1].lng);
+                                }
+                                distanceFromStart += trk.distance;
+                                point['distanceFromStart'] = distanceFromStart;
+                                point['distance'] += trk.distance;
+                            })
+                        } else {
+                            point.distanceFromStart = 0;
+                        }
+                    } else {
+                        track.hasOnlyTrk = true;
+                        t.points = Utils.getPointsDist(t.points);
+                        break;
+                    }
+                }
+            } else {
+                t.points = Utils.getPointsDist(t.points);
+            }
+        })
+    }
+}
+
+async function getGpxTrack(ctx) {
+    let trackData = {
+        tracks: ctx.selectedGpxFile.tracks,
+        wpts: ctx.selectedGpxFile.wpts,
+        metaData: ctx.selectedGpxFile.metaData,
+        ext: ctx.selectedGpxFile.ext,
+        analysis: null
+    }
+
+    if (!trackData.metaData.name) {
+        trackData.metaData.name = ctx.selectedGpxFile.name;
+    }
+
+    return await post(`${process.env.REACT_APP_GPX_API}/gpx/save-track-data`, trackData,
+        {
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        });
+}
+
+
+async function getNewGeometry(ctx, index) {
```

Comment:
- Probably should be renamed to updateRouteBetweenPoints or something similar
- getActivePoints - do we support disable points?